src/world/script_api/7E0E80.c

```diff
@@ -30,30 +145,96 @@ ApiStatus func_802803C8(ScriptInstance* script, s32 isInitialCall) {
     return ApiStatus_DONE2;
 }
 
+// Needs data migration
+#ifdef NON_MATCHING
+ApiStatus func_80280410(ScriptInstance* script, s32 isInitialCall) {
+    Shop* shop = gGameStatusPtr->mapShop;
+    s32 var1 = get_variable(script, *script->ptrReadPos);
+
+    if (!(shop->flags & 8)) {
+        shop->unk_08 = var1;
+        shop->flags |= 1;
+        func_800E98EC();
+        shop->unk_358 = 5;
+
+        if (gGameStatusPtr->pressedButtons & 0x8000) {
+            ScriptInstance* childScript;
+
+            disable_player_input();
+            disable_player_static_collisions();
+
+            childScript = start_script(&D_80284034_7E4EB4, 1, 0);
+            childScript->varTable[0] = var1;
+            D_80286520 = childScript;
+            D_80286524 = childScript->id;
+            shop->flags |= 8;
+            return ApiStatus_BLOCK;
+        } else {
+            return ApiStatus_DONE2;
+        }
+    } else if (does_script_exist(D_80286524)) {
+        return ApiStatus_BLOCK;
+    }
+
+    shop->flags &= ~0x8;
+    enable_player_static_collisions();
+    enable_player_input();
+    return ApiStatus_DONE2;
+}
+#else
 INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280410);
+#endif
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_8028051C);
+INCLUDE_ASM(s32, "world/script_api/7E0E80", ShowShopPurchaseDialog);
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280954);
+INCLUDE_ASM(s32, "world/script_api/7E0E80", shop_open_item_select_popup);
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280AC4);
+INCLUDE_ASM(s32, "world/script_api/7E0E80", shop_update_item_select_popup);
 
-void func_80280B44(void) {
+void shop_close_item_select_popup(void) {
     func_800F1538();
     func_800E9900();
     func_800E98C4();
     close_status_menu();
 }
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280B78);
+// Ordering issue
+#ifdef NON_MATCHING
+s32 shop_get_sell_price(s32 itemID) {
+    Shop* shop = gGameStatusPtr->mapShop;
+    s32 numItems = shop->numSpecialPrices;
+    s32 i;
+
+    for (i = 0; i < numItems; i++) {
+        if (shop->staticPriceList[i].itemID == itemID) {
+            return shop->staticPriceList[i].sellPrice;
+        }
+    }
+
+    return gItemTable[itemID].sellValue;
+}
+#else
+INCLUDE_ASM(s32, "world/script_api/7E0E80", shop_get_sell_price);
+#endif
+
+INCLUDE_ASM(s32, "world/script_api/7E0E80", ShowShopOwnerDialog);
+
+void shop_draw_item_name(s32 arg0, s32 posX, s32 posY) {
+    Shop* shop = gGameStatusPtr->mapShop;
+    StaticInventoryItem* siItem = &shop->staticInventory[shop->unk_08];
+    StaticItem* item = &gItemTable[siItem->unk_00];
 
-INCLUDE_ASM(s32, "world/script_api/7E0E80", func_80280BD0);
+    draw_msg(item->nameString, posX + 60 - (get_string_width(item->nameString, 0) >> 1), posY + 6, 255, 0, 0);
```

Comment:
`/ 2` instead of `>> 1`

Reply:
doesn't work

Reply:
wha, worrying. is `get_string_width`'s type `u32` perhaps?

Reply:
maybe, but casting it to a u32 doesn't work either. It does the wrong kind of shift